scipy/optimize/_minimize.py

```diff
@@ -601,54 +601,99 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
                       bounds=bounds, constraints=constraints,
                       callback=callback, **options)
 
-    if bounds is not None:
-        bounds = standardize_bounds(bounds, x0, meth)
-
     if constraints is not None:
         constraints = standardize_constraints(constraints, x0, meth)
 
+    remove_vars = False
+    if bounds is not None:
+        if meth.upper() in {"TNC", "SLSQP", "L-BFGS-B"}:
```

Comment:
We said we were going to leave 'trust-constr'` out of this since it's not having problems, right?
In any case, let's leave it out of an initial PR. We can deal with it later if we decide we need to.

"powell" was here before, but it doesn't use gradients, so it's really not needed. (Same with Nelder-Mead.)